File: Python/844. Backspace String Compare.py
class Solution:
    def backspaceCompare(self, S: str, T: str) -> bool:
        new_s = ""
        new_t = ""
        for i in range(len(S)):
            if S[i] == '#' and len(new_s) > 0:
                new_s = new_s[:-1]
            elif S[i] != '#':
                new_s += S[i]
        for i in range(len(T)):
            if T[i] == '#' and len(new_t) > 0:
                new_t = new_t[:-1]
            elif T[i] != '#':
                new_t += T[i]
        return new_s == new_t

File: 2023/844. Backspace String Compare.cpp
class Solution {
public:
    string getStr(string s) {
        string rs;
        int is = -1;
        for (char c : s) {
            if (c == '#') {
                if (is > -1) {
                    is--;
                }
            } else {
                if (is == -1) {
                    is = 0;
                }
                if (rs.size() <= is) {
                    rs += c;
                } else {
                    rs[is] = c;
                }
                is++;
            }
        }
        if (is <= 0) return "";
        return rs.substr(0, is);
    }
    bool backspaceCompare(string s, string t) {
        return getStr(s) == getStr(t);
    }
};